pkg/controllermanager/controller/project/stale/reconciler.go

```diff
@@ -149,24 +152,54 @@ func (r *Reconciler) projectInUseDueToBackupEntries(ctx context.Context, namespa
 }
 
 func (r *Reconciler) projectInUseDueToSecrets(ctx context.Context, namespace string) (bool, error) {
-	secretList := &corev1.SecretList{}
-	if err := r.Client.List(
-		ctx,
-		secretList,
-		client.InNamespace(namespace),
-		gardenerutils.UncontrolledSecretSelector,
-		client.MatchingLabels{v1beta1constants.LabelSecretBindingReference: "true"},
-	); err != nil {
+	getSecrets := func(matchKeyLabel string) (*corev1.SecretList, error) {
+		secretList := &corev1.SecretList{}
+		err := r.Client.List(
+			ctx,
+			secretList,
+			client.InNamespace(namespace),
+			gardenerutils.UncontrolledSecretSelector,
+			client.MatchingLabels{matchKeyLabel: "true"},
+		)
+		return secretList, err
+	}
+
+	secretBindingRefSecretList, err := getSecrets(v1beta1constants.LabelSecretBindingReference)
+	if err != nil {
+		return false, err
+	}
+
+	secretBindingSecretNames := computeSecretNames(secretBindingRefSecretList.Items)
```

Comment:
Hi @timuthy,

Maybe you wanted to cc @shafeeqes as the `v1beta1constants.LabelSecretBindingReference` label was introduced with https://github.com/gardener/gardener/commit/11270b0f702fb2dd7af28e809796d433e1c139ba.

Anyways, what is your idea for eliminating the `computeSecretNames` func?

Reply:
Thanks @ialidzhikov!

> Anyways, what is your idea for eliminating the computeSecretNames func?

The idea is to get rid of potential stale code.

IIUC, this function sorted out secrets that are managed by Gardener, i.e. have a shoot owner ref. Since some time we have the label selector `reference.gardener.cloud/secretbinding` in place, and I wonder which secrets are left that need to be filtered out here?!

Reply:
/ping @dimityrmirchev